src/tego_ui/main.cpp

```diff
@@ -273,7 +279,67 @@ static bool initSettings(SettingsFile *settings, QLockFile **lockFile, QString &
     if (args.size() > 1) {
         configPath = args[1];
     } else {
+        // TODO: remove this profile migration after sufficient time has passed (EOY 2021)
+        auto legacyConfigPath = []() -> QString {
+            QString configPath;
+#ifdef Q_OS_MAC
+            if (!qApp->applicationDirPath().contains(QStringLiteral("/Applications"))) {
+                // Try old configuration path first
+                configPath = appBundlePath() + QStringLiteral("config.torsion");
+                if (!QFile::exists(configPath))
+                    configPath = appBundlePath() + QStringLiteral("config.ricochet");
+            }
+#else
+            configPath = qApp->applicationDirPath() + QStringLiteral("/config");
+#endif
+            return configPath;
+        }();
         configPath = QStandardPaths::writableLocation(QStandardPaths::AppLocalDataLocation);
+
+        logger::println("configPath : {}", configPath);
+        logger::println("legacyConfigPath : {}", legacyConfigPath);
+
+        // only put up migration UX when
+        if (// old path differs from new path
+            configPath != legacyConfigPath &&
+            // the old path exists
+            QFile::exists(legacyConfigPath) &&
+            // the new path does not exist
+            !QFile::exists(configPath)) {
+
+            QMessageBox msgBox;
+            msgBox.setWindowTitle(QStringLiteral("Profile Migration"));
+            msgBox.setText(QStringLiteral("Ricochet Refresh has detected an existing legacy profile. Do you want to import it?"));
+            msgBox.setIcon(QMessageBox::Question);
+            msgBox.setDetailedText(
+                QStringLiteral(
+                    "Previous versions of Ricochet Refresh stored your profile data in the application's install location. If you import your legacy profile, it will be moved to a new location within your home directory.\n\n"
+                    "Old profile: '%1'\n"
+                    "New profile: '%2'").arg(legacyConfigPath).arg(configPath));
+            msgBox.setStandardButtons(QMessageBox::Yes | QMessageBox::No | QMessageBox::Abort);
+            msgBox.setDefaultButton(QMessageBox::Abort);
+
+            switch(static_cast<QMessageBox::StandardButton>(msgBox.exec()))
+            {
+                case QMessageBox::Yes:
+                    // migrate profile
+                    if(!QDir().rename(legacyConfigPath, configPath)) {
```

Comment:
Hmm this may be a wider issue, but if a config is invalid then we throw an error on [line 377](https://github.com/pospeselr/ricochet-refresh/blob/issue_89/src/tego_ui/main.cpp#L377) without prompting the user to create a new config. We should probably check here if the config is valid before moving it to the config dir, or better yet, instead of just throwing an error on an invalid config, we prompt the user to create a new config.

Reply:
So there are basically two error conditions here: lock-file errors, and load/parse errors. In the case of a lock file preventing forward progress, this will already be implicitly detected by QDir::rename(); we won't be able to move the profile directory if its lock-file is not currently writable. Finally, if we make it down actually reading the configuration and run into some sort of parse error, that also isn't *really* our problem since the profile would fail to load even before this migration change.